### ISSUE TITLE
[Bug] Missing Extensions Options in UnoFeatures starting from Blank Preset when creating projects with Uno Templates, VS Wizard, or Live Wizard

### ISSUE THREAD

**Issue by agneszitte** [unoplatform/uno.templates#991]
### Description:
I'm encountering an issue with the latest Uno templates (**v5.4.5**) and the latest VS Uno Extension (**v5.4.11**) with Wizard when attempting to add various extensions options to a blank preset project. The issue affects not just **Localization**, but other extension options such as **HTTP** as well, which do not get properly reflected in the `<UnoFeatures />` in the .csproj of the project. Below are the details with the **Localization** option as an example :

#### Using Visual Studio Wizard:
1. Select a **Blank Preset** project.
2. Enable **Dependency Injection (DI)** and **Localization**.
3. Create the project.
4. **Observed Behavior:** The corresponding entry **Localization** is missing from `UnoFeatures` in the `.csproj` file of the project.

   https://github.com/user-attachments/assets/edeb1c4f-8c38-48f7-bafc-7286dbe297f4

#### Using Live Wizard:
1. Select a **Blank Preset** project.
2. Enable **Dependency Injection (DI)** and **Localization**.
3. At the end, notice that option `-loc` is missing from the generated command.
4. Copy and paste the command to manually create the project.
5. **Observed Behavior:** Obviously, no corresponding entry **Localization** in `UnoFeatures` in the `.csproj` file of the project.

    https://github.com/user-attachments/assets/0070cbc4-e649-40ce-973f-7767eee95ecf

#### Using CLI:
1. Use the blank preset command (`dotnet new unoapp -preset=blank -o test`) and manually add `-di` and the localization option `-loc`.
2. **Observed Behavior:** The corresponding entry **Localization** is missing from `UnoFeatures` in the `.csproj` file of the project.

    https://github.com/user-attachments/assets/db245fcd-b6af-4dae-9dc4-2ff87acff9e7

### Additional Observations:
- **Additional Live Wizard Tests:**
   - Blank preset + DI produces a correct command.
> [!WARNING] 
> Adding extensions such as **Configuration** or **HTTP** or **Localization** after selecting DI does not update the command at all, meaning that these options are not included in the generated project.
   
https://github.com/user-attachments/assets/a9e3dbed-c033-40e8-a0c2-ce64a4338fe4

### Expected Behavior:
In all cases, when selecting additional extensions such as **Localization**, **Configuration**, or **HTTP**, these options should be properly reflected in the `UnoFeatures` of the `.csproj` file and the generated command (for Live Wizard).

**Comment by agneszitte** [comment]
Needs to be verified if if this is a regression or if it’s been there for a while but just went unnoticed.
Are we missing some additional tests?

**Comment by agneszitte** [comment]
Are we missing any additional tests?

  From @jeromelaban: 
  > If we do not have such validations in the uno.templates generation, we need to add some more.

**Comment by agneszitte** [comment]
cc @nickrandolph for info

**Comment by eriklimakc** [comment]
`Localization` and `Http` depend on `Configuration`, so a file in uno.templates was changed so that this dependency was added (https://github.com/unoplatform/uno.templates/pull/1128).

**Comment by DevTKSS** [comment]
hey there, since this is quite the same I did recognize (but after creation time) I would give some additional detail into consideration:
created the blanc project, decided I want to use http (TheCatAPI Sample Tutorial using) which made me add configuration also because I dont want the api-key to be clear typed in my solution general files (still looking for a kind of vault for this in desktop uno apps which are not needing aws or azure?) so added <UnoFeatures> Tag into the project, reopened vs 2022 community edition and after checking all things it seems to be the case that beside the Namespace Uno.Extensions .... which is mentioned in the guides, for DI we have to manually add the Microsoft.Extensions.Hosting Namespace?
@agneszitte seen you also used here DI which is what I tryed to do same just without the cmd line but along the manuals with the tags, so maybe there is also a kind of dependency between them both problems?
Kind regards

**Comment by agneszitte** [comment]
@eriklimakc can you help summarize if there is anything left to fix for this issue please

**Comment by Jen-Uno** [comment]
@erikvilima  is there something left to do for this issue?

**Comment by erikvilima** [comment]
@Jen-Uno @agneszitte Yes, we can improve our docs based on this comment https://github.com/unoplatform/uno.templates/issues/991#issuecomment-2520073558.

We can instruct users to add the `Microsoft.Extensions.Hosting` namespace to GlobalUsings.cs on the Hosting docs page https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Hosting/HowTo-HostingSetup.html

Also we could link that same page to the [DI page](https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/DependencyInjection/HowTo-DependencyInjectionSetup.html) to instruct users on how to setup hosting to use DI in case they create an app without it.

**Comment by Jen-Uno** [comment]
@erikvilima please create a new task for the documentation and, if you have the time, include it in this fix. Thanks!

**Comment by booskander** [comment]
<img width="260" height="297" alt="Image" src="https://github.com/user-attachments/assets/f13c9e28-218d-4a2f-8f88-bd0f6751393f" />

It seems to be fixed on my side, Localization is present inside the .csproj

I could not indeed select Localization without also having Configuration selected which corresponds to the desired behaviour!
